src/etcd/model/kv.cr

```diff
@@ -23,20 +24,18 @@ module Etcd::Model
     getter kvs : Array(Etcd::Model::Kv) = [] of Etcd::Model::Kv
   end
 
-  struct PutResponse < Base
-    getter header : Header
+  struct PutResponse < WithHeader
+    # Why is this nillable
     getter prev_kv : Kv?
```

Comment:
Nillable because the previous is only included if you ask for the previous kv.

Reply:
And it might not be present

Reply:
Theres no nillable type marking on the openapi specs, makes everything a bit more confusing than it should be

Reply:
I think you should work against the API rather than the spec. I remember it not being consistent.